projects/packages/jetpack-mu-wpcom/src/features/wpcom-admin-interface/wpcom-admin-interface.php

```diff
@@ -273,3 +276,47 @@ function wpcom_show_admin_interface_notice() {
 	);
 }
 add_action( 'admin_notices', 'wpcom_show_admin_interface_notice' );
+
+/**
+ * Fetches the current assignment to the duplicate views experiment.
+ *
+ * @return string
+ */
+function wpcom_fetch_duplicate_views_assignment() {
+	// TODO: We don't know yet the experiment name.
+	$experiment_platform = 'calypso';
+	$experiment_name     = "{$experiment_platform}_duplicate_views_placeholder";
+
+	if ( ( new Host() )->is_wpcom_simple() ) {
+		return \ExPlat\assign_current_user( $experiment_name );
+	} else {
+		// If there isn't a WP.com connection, we should keep users in WP Admin.
+		if ( ! ( new Jetpack_Connection() )->is_user_connected() ) {
+			return 'treatment';
+		}
+
+		$request_path = add_query_arg(
+			array( 'experiment_name' => $experiment_name ),
+			"/experiments/0.1.0/assignments/{$experiment_platform}"
+		);
+		$response     = Client::wpcom_json_api_request_as_user( $request_path, 'v2' );
```

Comment:
This might be problematic from performance PoV since there might be a lot of calls to this function in the context of a single request.